### PR TITLE
Adds support for standalone applications

### PR DESCRIPTION
**Pull request by jkotalik**
For #251

### CODE REVIEW

File: src/AspNetCore/Src/inprocessapplication.cxx [hunk 1]
```diff
@@ -684,3 +698,83 @@ IN_PROCESS_APPLICATION::FindHighestDotNetVersion(
...
+    }
+
+    struHostfxrPath.QueryStr()[dwPosition] = L'\0';
+    struApplicationDllPath.QueryStr()[dwPosition] = L'\0';
+
+    // Whenever we directly modify the stru buffer, we need to call SyncWithBuffer
+    if (FAILED(hr = struHostfxrPath.SyncWithBuffer())
+        || FAILED(hr = struHostfxrPath.Append(L"\\hostfxr.dll"))
+        || FAILED(hr = struApplicationDllPath.SyncWithBuffer())
+        || FAILED(hr = struApplicationDllPath.Append(L"\\NativeIISSample.dll")))
```
Comment (Tratcher) [inline src/AspNetCore/Src/inprocessapplication.cxx]: ?

Comment (jkotalik) [reply src/AspNetCore/Src/inprocessapplication.cxx]: Oops, forgot to actually parse that out XD

File: src/AspNetCore/Src/inprocessapplication.cxx [hunk 2]
```diff
@@ -684,3 +698,83 @@ IN_PROCESS_APPLICATION::FindHighestDotNetVersion(
...
+//   hostfxr.dll
+//   Application.exe
+//   Application.dll
+//   etc.
+// We get the full path to hostfxr.dll and Application.dll and run hostfxr_main,
+// passing in Application.dll.
+// Assuming we don't need Application.exe as the dll is the actual application.
+//
+HRESULT
+IN_PROCESS_APPLICATION::RunStandaloneApplication()
```
Comment (davidfowl) [inline src/AspNetCore/Src/inprocessapplication.cxx]: I would refactor the code to find the correct module and then from there share the execution logic. Seems like a bunch of copy pasta here.

We should also document that standalone applications when running the in process mode *does not* pick up muxer changes. This means any changes inside of `dotnet.exe` itself won't show up.

**Comment by jkotalik** [comment]
🆙 📅

File: src/AspNetCore/Src/inprocessapplication.cxx [hunk 3]
```diff
@@ -684,3 +560,239 @@ IN_PROCESS_APPLICATION::FindHighestDotNetVersion(
...
+// passing in Application.dll.
+// Assuming we don't need Application.exe as the dll is the actual application.
+//
+HRESULT
+IN_PROCESS_APPLICATION::GetStandaloneApplicationArguments(
+    STRU* struApplicationDllPath,
+    STRU* struHostfxrPath
+)
+{
+    HRESULT     hr = S_OK;
```
Comment (davidfowl) [inline src/AspNetCore/Src/inprocessapplication.cxx:582]: I don't even understand this formatting.

Comment (jkotalik) [reply src/AspNetCore/Src/inprocessapplication.cxx:582]: It's the standard across the project, as odd as it looks.

Comment (davidfowl) [reply src/AspNetCore/Src/inprocessapplication.cxx:582]: That's awful.

**Comment by jkotalik** [comment]
Closing as the refactor will take a different approach in loading hostfxr and the application path.

**Pull request closed without merging**